bids/analysis/transformations/base.py

```diff
@@ -351,13 +349,18 @@ def compare_variables(a, b):
                 fc = get_col_data(variables[0])
                 if not all([compare_variables(fc, get_col_data(c))
                             for c in variables[1:]]):
-                    msg = "Misaligned sparse variables found."
-                    if force:
-                        msg += (" Forcing all sparse variables to dense in "
+                    if self._force_dense_align:
+                        msg = ("Forcing all sparse variables to dense in "
                                 "order to ensure proper alignment.")
                         sr = self.collection.sampling_rate
                         variables = [c.to_dense(sr) for c in variables]
-                    warnings.warn(msg)
+                        warnings.warn(msg)
+                    else:
+                        raise ValueError(
+                            "Misaligned sparse variables found."
+                            "To force variables into alignment by densifying,"
+                            "set dense=True in the Transformation arguments"
+                            )
```

Comment:
If we have the variable names, it would be helpful to include them in any exceptions.

Reply:
Well, we don't have the names of which ones don't match, just that there is at least one mismatch. There really isn't a "ground truth"  so it's hard to say which variables are the "correct" and which are the "incorrect" ones, just that they are not all in the same alignment.

Reply:
Sure, the point is just that when this pops up, it's unlikely to be in a context where it's easy to see the names of the mismatching columns from the rest of the traceback.

Reply:
I agree it's not ideal, but I think the alternative would be to print all the columns. Either way, I think just saying dense=True is what most users would want to do in this situation.